### PR TITLE
Adding disconnect reason

### PR DESCRIPTION
**Pull request by davidzhao**
When a participant is disconnected, we have the reason for disconnection.

Pass through this event to FFI clients so they could determine why a RemoteParticipant had disconnected

### CODE REVIEW

File: livekit-api/src/services/sip.rs [hunk 1]
```diff
@@ -58,18 +60,24 @@ pub struct CreateSIPTrunkOptions {
 #[derive(Default, Clone, Debug)]
 pub struct CreateSIPInboundTrunkOptions {
     /// Optional free-form metadata.
-    pub metadata: String,
+    pub metadata: Option<String>,
```
Comment (davidzhao) [inline livekit-api/src/services/sip.rs:63]: these aren't required, updating them to optional

File: livekit/src/room/mod.rs [hunk 2]
```diff
@@ -783,6 +783,7 @@ impl RoomSession {
             let remote_participant = self.get_participant_by_sid(&participant_sid);
             if pi.state == proto::participant_info::State::Disconnected as i32 {
                 if let Some(remote_participant) = remote_participant {
+                    remote_participant.update_info(pi.clone());
```
Comment (davidzhao) [inline livekit/src/room/mod.rs:786]: even when the remote participant is disconnected, we have to update it to keep its reason

File: livekit-api/src/services/sip.rs (hunk 1 above)
Comment (theomonnom) [reply livekit-api/src/services/sip.rs:63]: Generally using default values is fine in this case, but an Option is ok too

Comment (davidzhao) [reply livekit-api/src/services/sip.rs:63]: yeah, it's easier for users if they didn't have to fill out every field as default

**Pull request merged**